File: src/java/magmac/app/lang/java/node/Parameters.java
package magmac.app.lang.java.node;

import magmac.api.collect.list.Lists;
import magmac.app.compile.error.CompileResult;
import magmac.app.compile.node.Node;
import magmac.app.compile.rule.NodeListRule;
import magmac.app.compile.rule.OrRule;
import magmac.app.compile.rule.Rule;
import magmac.app.lang.ValueFolder;
import magmac.app.lang.java.Deserializers;

public class Parameters {
    public static CompileResult<Parameter> deserialize(Node node) {
        return Deserializers.orError("parameter", node, Lists.of(
                Deserializers.wrap(Whitespace::deserialize),
                Deserializers.wrap(Definition::deserialize)
        ));
    }

    public static NodeListRule createParametersRule(Rule definition) {
        return new NodeListRule("parameters", new ValueFolder(), new OrRule(Lists.of(
                Whitespace.createWhitespaceRule(),
                definition
        )));
    }
}

File: src/web/magmac/app/lang/node/Parameters.ts
import { Lists } from "../../../../magmac/api/collect/list/Lists";
import { CompileResult } from "../../../../magmac/app/compile/error/CompileResult";
import { Node } from "../../../../magmac/app/compile/node/Node";
import { NodeListRule } from "../../../../magmac/app/compile/rule/NodeListRule";
import { OrRule } from "../../../../magmac/app/compile/rule/OrRule";
import { Rule } from "../../../../magmac/app/compile/rule/Rule";
import { JavaDeserializers } from "../../../../magmac/app/lang/java/JavaDeserializers";
import { JavaRules } from "../../../../magmac/app/lang/JavaRules";
import { ValueFolder } from "../../../../magmac/app/lang/ValueFolder";
import { Deserializers } from "../../../../magmac/app/lang/Deserializers";
import { JavaParameter } from "../../../../magmac/app/lang/java/JavaParameter";
export class Parameters {
	deserialize(node : Node) : CompileResult<JavaParameter> {return 0.orError( "parameter", 0, 0.of( 0.wrap( 0.deserializeWhitespace), 0.wrap( 0.deserializeTypedDefinition)));;}
	createParametersRule(definition : Rule) : Rule {return 0.createNodeListRule( "parameters", new ValueFolder( ), new OrRule( 0.of( 0.createWhitespaceRule( ), 0)));;}
}
